### ISSUE TITLE
Possibility of using the label () or singularLabel () instead of the title

### ISSUE THREAD

**Issue by GabrielApG** [chris-ware/nova-breadcrumbs#4]
Have you thought about possibility of using the label() or singularLabel() instead of the title ?

**Comment by nasrulhazim** [comment]
Yeah, that will be good. At the moment can overwrite the behaviour:

```php
protected function serializeWithId(Collection $fields)
    {
        $parent = parent::serializeWithId($fields);

        return array_merge($parent, [
            'title' => $this->singularLabel(), // $this->label()
        ]);
    }
```

**Comment by bernhardh** [comment]
@nasrulhazim suggestion is, as @chris-ware said, not really usefull, since `serializeWithId` method is for a single resource item, not the resource "collection".

But I think there is still an issue with the breadcrumb:

![image](https://user-images.githubusercontent.com/642292/49737268-ae9dd380-fc8c-11e8-9603-eafcd40c8aa3.png)

In the Breadcrumb there is "Home / City Addresses" but it should be "Home / Locations". The resource is called `CityAddresses" but the human readable form, which is used in the sidebar and the titles (see H1 of screenshot) comes from label/singularLabel.

**Comment by nasrulhazim** [comment]
I think we can create another method:

```php
public $breadcrumb_label = 'Locations';

public function breadcrumbLabel()
{
    if(property_exists($this, 'breadcrumb_label') {
        return $this->breadcrumb_label;
   }

    return $this->title();
}
```

```php
protected function serializeWithId(Collection $fields)
    {
        $parent = parent::serializeWithId($fields);

        return array_merge($parent, [
            'title' => $this->breadcrumbLabel(),
        ]);
    }
```

With above setup, the usage of `$this->title()` still in use.

**Comment by chris-ware** [comment]
@nasrulhazim I still think you may be misunderstanding the title() function's purpose. It only returns the title of the resource. The above from @bernhardh is solved by using the resource name from the resource itself. I will be coding this and releasing this later today.

**Comment by chris-ware** [comment]
I've added this functionality into the latest version of master, that will be released as a new minor version.

**Comment by chris-ware** [comment]
Closing issue as this is now implemented in version 0.5.0
